core/store/src/trie/mem/arena/concurrent.rs

```diff
@@ -0,0 +1,252 @@
+use super::alloc::{allocation_class, allocation_size, CHUNK_SIZE};
```

Comment:
Very high level question about the logic here: why do we call it "concurrent"? Usually I expect some concurrency logic to be there, however here it seem to mean that this is just `ArenaWithoutDeallocation`, or add-only arena. I think renaming logic to that would help by not introducing additional usage context.

With that in mind, could we also reuse existing Arena/ArenaMemory impls, by just not implementing deallocs?

Reply:
It's concurrent because we allow multiple threads to allocate at the same time, but sharing the "arena-logical" memory. We cannot just reuse the ArenaMemory impl, because this sharing of arena-logical memory requires an additional "address translation table" in the ConcurrentArenaMemory. You could argue that ArenaMemory could just be implemented by having this table be the identity lookup table, but that's additional memory overhead on every dereference.